src/dom-to-image-more.js

```diff
@@ -1192,22 +1192,77 @@
-                // computed styles. Elements must be rendered in order to query their computed styles. The <iframe> won't
-                // render at all with `display: none`, so we have to use `visibility: hidden` with `position: fixed`.
-                sandbox = document.createElement('iframe');
-                sandbox.style.visibility = 'hidden';
-                sandbox.style.position = 'fixed';
-                document.body.appendChild(sandbox);
-                // Ensure that the iframe is rendered in standard mode
-                const charset = document.createElement('meta');
-                charset.setAttribute('charset', document.characterSet || 'UTF-8');
-                sandbox.contentDocument.head.appendChild(charset);
-                sandbox.contentDocument.title = 'sandbox';
+            if (sandbox) {
+                return sandbox.contentWindow;
             }
 
-            return sandbox.contentWindow;
+            // figure out how this document is defined (doctype and charset)
+            const charsetToUse = document.characterSet || 'UTF-8';
+            const docType = document.doctype;
+            const docTypeDeclaration = docType
+                ? `<!DOCTYPE ${escapeHTML(docType.name)} ${escapeHTML(
+                      docType.publicId
+                  )} ${escapeHTML(docType.systemId)}`.trim() + '>'
+                : '';
+
+            // Create a hidden sandbox <iframe> element within we can create default HTML elements and query their
+            // computed styles. Elements must be rendered in order to query their computed styles. The <iframe> won't
+            // render at all with `display: none`, so we have to use `visibility: hidden` with `position: fixed`.
+            sandbox = document.createElement('iframe');
+            sandbox.id = 'domtoimage-sandbox-' + util.uid();
+            sandbox.style.visibility = 'hidden';
+            sandbox.style.position = 'fixed';
+            document.body.appendChild(sandbox);
+
+            return tryTechniques(
+                sandbox,
+                docTypeDeclaration,
+                charsetToUse,
+                'domtoimage-sandbox'
+            );
+
+            function escapeHTML(unsafeText) {
+                if (unsafeText) {
+                    const div = document.createElement('div');
+                    div.innerText = unsafeText;
+                    return div.innerHTML;
+                } else {
+                    return '';
+                }
+            }
+
+            function tryTechniques(sandbox, doctype, charset, title) {
+                // try the good old-fashioned document write with all the correct attributes set
+                try {
+                    sandbox.contentWindow.document.write(
+                        `${doctype}<html><head><meta charset='${charset}'><title>${title}</title></head><body></body></html>`
```

Comment:
Looks good to me! The only thing I'd recommend is using double quotes inside HTML, i.e. `charset="${charset}"` since that is considered best practice (this could be done later though - it's not important).